wiring/inc/spark_wiring_spi.h

```diff
@@ -228,14 +228,14 @@ class SPIClass {
   void lock()
   {
 #if PLATFORM_THREADING
-    mutex_.lock();
+    HAL_SPI_Acquire(_spi, NULL);
```

Comment:
This should also be conditional on `HAL_PLATFORM_SPI_HAL_THREAD_SAFETY`.

Reply:
Now that the Spark Core platform has been removed from the code base, does `#if PLATFORM_THREADING` only block GCC, platform id 3?

Can `PLATFORM_THREADING` just be swapped with `HAL_PLATFORM_SPI_HAL_THREAD_SAFETY`?

Reply:
Yup.